lib/params/RenderParams.cpp

```diff
@@ -497,6 +497,39 @@ void RenderParams::SetFieldVariableNames(vector<string> varnames){
 //	setAllBypass(false);
 }
 
+std::string RenderParams::GetXFieldVariableName() const {
+    std::vector< std::string> fieldVars = GetFieldVariableNames();
```

Comment:
It would be a good idea to assert in all of these setters and getters that fieldVars is of sufficient length before indexing into it.

Reply:
Done